### PR TITLE
fix(unlock-app): Handle an edge case where QR code urls have double encoded the data and missing tokenId

### PR DESCRIPTION
**Pull request by searchableguy**
This is a fix for QR codes generated before june.

1. There was a bug in the past which caused data to be double encoded in the URI. We fixed this sometime later but since some QR codes were generated through that bug, this fixes it by double decoding which shouldn't cause problem even if URI was only once encoded.
2. Similar to this, our old QR codes didn't include tokenId. I've made it optional.

# Description

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
We can either merge this or don't handle the problematic past QR codes.

**Comment by searchableguy** [comment]
Since we do warn people to generate QR codes if it's been a while.

**Comment by julien51** [comment]
That's great. I'll approve but we should probably add comments to make sure we can clean that up in a few months.

File: unlock-app/src/components/interface/VerificationStatus.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const VerificationStatus = ({ config, onVerified, onClose }: Props) => {
   const { isLoading: isKeyLoading, data: key } = useQuery(
     [network, tokenId, lockAddress],
     async () => {
-      if (lockVersion && lockVersion >= 10) {
+      if (lockVersion && lockVersion >= 10 && tokenId) {
```
Comment (julien51) [inline unlock-app/src/components/interface/VerificationStatus.tsx:64]: Some older QR codes might have been
```suggestion
      // Some older QR codes might have been generated without a tokenId in the payload. Clean up after January 2023
      if (lockVersion && lockVersion >= 10 && tokenId) {
```

File: unlock-app/src/utils/verification.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ interface Options {
 export function getMembershipVerificationConfig({ data, sig }: Options) {
   try {
     if (sig && data) {
-      const raw = decodeURIComponent(data)
+      const raw = decodeURIComponent(decodeURIComponent(data))
```
Comment (julien51) [inline unlock-app/src/utils/verification.ts:27]: Will that create issues if the payload is not encoded twice?

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
Lgtm

**Comment by searchableguy** [comment]
@kalidiagne please test on your end as well and report if it doesn't break anything.

**Review by kalidiagne — APPROVED** [review]

**Comment by kalidiagne** [comment]
> @kalidiagne please test on your end as well and report if it doesn't break anything.

Works fine!

**Pull request merged**